diff -ru CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -217,11 +217,11 @@
 		-DGENERATEDSAXPARSER_XMLPARSER_LIBXML
 		-DXMLPARSER_LIBXML
 	)
-	find_package(LibXml2)
+	#find_package(LibXml2)
 	if (LIBXML2_FOUND)
 		message(STATUS "SUCCESSFUL: LibXml2 found")
 	else ()  # if xml2 not found building its local copy from ./Externals
-		if (WIN32)
+		if (1)
			message("WARNING: Native LibXml2 not found, taking LibXml from ./Externals")
			add_subdirectory(${EXTERNAL_LIBRARIES}/LibXML)
			set(LIBXML2_INCLUDE_DIR
@@ -254,11 +254,11 @@
 endif()
 
 #adding PCRE
-find_package(PCRE)
+#find_package(PCRE)
 if (PCRE_FOUND)
 	message(STATUS "SUCCESSFUL: PCRE found")
 else ()  # if pcre not found building its local copy from ./Externals
-	if (WIN32 OR APPLE)
+	if (1)
 		message("WARNING: Native PCRE not found, taking PCRE from ./Externals")
 		add_definitions(-DPCRE_STATIC)
 		add_subdirectory(${EXTERNAL_LIBRARIES}/pcre)
diff -ru DAEValidator/library/include/no_warning_begin DAEValidator/library/include/no_warning_begin
--- DAEValidator/library/include/no_warning_begin
+++ DAEValidator/library/include/no_warning_begin
@@ -2,6 +2,9 @@
 #if defined(_WIN32)
 #	pragma warning(push)
 #	pragma warning(disable:4668)
+#	if _MSC_VER >=1900
+#		pragma warning(disable:5031)
+#	endif
 #	if defined(_MSC_VER) && defined(_DEBUG)
 #		pragma warning(disable:4548)
 #	endif
diff -ru GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
--- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
+++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
@@ -1553,7 +1553,7 @@
 #if defined(COLLADABU_OS_WIN) && !defined(__MINGW32__)
         return _isnan( value ) ? true : false;
 #else
-#ifdef isnan
+#if defined(isnan) || defined(__APPLE__)
         return isnan( value );
 #else
         return std::isnan(value);
diff -ru DAEValidator/library/src/ArgumentParser.cpp DAEValidator/library/src/ArgumentParser.cpp
--- DAEValidator/library/src/ArgumentParser.cpp
+++ DAEValidator/library/src/ArgumentParser.cpp
@@ -6,10 +6,10 @@
 
 using namespace std;
 
-#ifdef _MSC_VER
-#define NOEXCEPT _NOEXCEPT
-#else
+#ifndef _NOEXCEPT
 #define NOEXCEPT noexcept
+#else
+#define NOEXCEPT _NOEXCEPT
 #endif
 
 namespace opencollada
