diff -ru CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -217,11 +217,11 @@
 		-DGENERATEDSAXPARSER_XMLPARSER_LIBXML
 		-DXMLPARSER_LIBXML
 	)
-	find_package(LibXml2)
+	#find_package(LibXml2)
 	if (LIBXML2_FOUND)
 		message(STATUS "SUCCESSFUL: LibXml2 found")
 	else ()  # if xml2 not found building its local copy from ./Externals
-		if (WIN32)
+		if (1)
			message("WARNING: Native LibXml2 not found, taking LibXml from ./Externals")
			add_subdirectory(${EXTERNAL_LIBRARIES}/LibXML)
			set(LIBXML2_INCLUDE_DIR
@@ -254,11 +254,11 @@
 endif()
 
 #adding PCRE
-find_package(PCRE)
+#find_package(PCRE)
 if (PCRE_FOUND)
 	message(STATUS "SUCCESSFUL: PCRE found")
 else ()  # if pcre not found building its local copy from ./Externals
-	if (WIN32 OR APPLE)
+	if (1)
 		message("WARNING: Native PCRE not found, taking PCRE from ./Externals")
 		add_definitions(-DPCRE_STATIC)
 		add_subdirectory(${EXTERNAL_LIBRARIES}/pcre)
diff -ru DAEValidator/library/include/no_warning_begin DAEValidator/library/include/no_warning_begin
--- DAEValidator/library/include/no_warning_begin
+++ DAEValidator/library/include/no_warning_begin
@@ -2,6 +2,9 @@
 #if defined(_WIN32)
 #	pragma warning(push)
 #	pragma warning(disable:4668)
+#	if _MSC_VER >=1900
+#		pragma warning(disable:5031)
+#	endif
 #	if defined(_MSC_VER) && defined(_DEBUG)
 #		pragma warning(disable:4548)
 #	endif
diff -ru GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
--- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
+++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
@@ -1553,7 +1553,7 @@
 #if defined(COLLADABU_OS_WIN) && !defined(__MINGW32__)
         return _isnan( value ) ? true : false;
 #else
-#ifdef isnan
+#if defined(isnan) || defined(__APPLE__)
         return isnan( value );
 #else
         return std::isnan(value);
diff -ru DAEValidator/library/src/ArgumentParser.cpp DAEValidator/library/src/ArgumentParser.cpp
--- DAEValidator/library/src/ArgumentParser.cpp
+++ DAEValidator/library/src/ArgumentParser.cpp
@@ -6,10 +6,10 @@
 
 using namespace std;
 
-#ifdef _MSC_VER
-#define NOEXCEPT _NOEXCEPT
-#else
+#ifndef _NOEXCEPT
 #define NOEXCEPT noexcept
+#else
+#define NOEXCEPT _NOEXCEPT
 #endif
 
 namespace opencollada
diff --git common/libBuffer/include/CommonFWriteBufferFlusher.h common/libBuffer/include/CommonFWriteBufferFlusher.h
--- common/libBuffer/include/CommonFWriteBufferFlusher.h
+++ common/libBuffer/include/CommonFWriteBufferFlusher.h
@@ -13,32 +13,7 @@
 
 #include "CommonIBufferFlusher.h"
 
-#if (defined(WIN64) || defined(_WIN64) || defined(__WIN64__)) || (defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__))
-#if defined(__GNUC__) && !defined(_LIBCPP_VERSION)
-#	include <tr1/unordered_map>
-#else
-#	include <unordered_map>
-#endif
-#else
-#	include <tr1/unordered_map>
-#endif
-
-#ifdef _LIBCPP_VERSION
-// If we're compiling with libc++, create a namespace alias for tr1 that points to std.
-// Not particularly elegant, and largely should be filed under "hack", but it works for OS X with clang for now.
-namespace std {
-    namespace tr1 = std;
-}
-#endif
-
-/* size_t for gcc, may want to move this include some place else - campbell */
-#ifdef __GNUC__
-#   include <cstdlib> /* size_t */
-#   include <cstdio>  /* FILE */
-#ifdef __FreeBSD__
-#include <stdint.h> /* int64_t */
-#endif
-#endif
+#include <unordered_map>
 
 #if __cplusplus > 199711L
 typedef int64_t __int64;
@@ -58,7 +33,8 @@ namespace Common
 #else
 		typedef __int64 FilePosType;
 #endif
-	typedef std::tr1::unordered_map<MarkId, FilePosType > MarkIdToFilePos;
+
+	typedef std::unordered_map<MarkId, FilePosType> MarkIdToFilePos;
 
 	public:
 		static const size_t DEFAUL_BUFFER_SIZE = 64*1024;
