diff --git a/openvdb/CMakeLists.txt b/openvdb/CMakeLists.txt
index 9404dbca..3d6de615 100644
--- a/openvdb/CMakeLists.txt
+++ b/openvdb/CMakeLists.txt
@@ -78,10 +78,13 @@ SET ( OPENVDB_PATCH_VERSION_NUMBER ${_openvdb_patch_version_number} CACHE STRING
 # MESSAGE ( "OPENVDB_MINOR_VERSION_NUMBER = ${OPENVDB_MINOR_VERSION_NUMBER}" )
 # MESSAGE ( "OPENVDB_PATCH_VERSION_NUMBER = ${OPENVDB_PATCH_VERSION_NUMBER}" )
 
-FIND_PACKAGE ( Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS iostreams system thread ${OPENVDB_BOOST_PYTHON_LIBRARY_COMPONENT_NAME} )
+FIND_PACKAGE ( Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS iostreams thread ${OPENVDB_BOOST_PYTHON_LIBRARY_COMPONENT_NAME} )
 IF (WIN32 AND OPENVDB_DISABLE_BOOST_IMPLICIT_LINKING)
   ADD_DEFINITIONS ( -DBOOST_ALL_NO_LIB )
 ENDIF ()
+if(WIN32)
+ADD_DEFINITIONS ( -D__TBB_NO_IMPLICIT_LINKAGE )
+endif()
 
 FIND_PACKAGE ( Blosc REQUIRED )
 FIND_PACKAGE ( TBB REQUIRED )
@@ -182,7 +185,6 @@ ADD_LIBRARY ( openvdb_shared SHARED
 
 TARGET_LINK_LIBRARIES ( openvdb_static
   ${Boost_IOSTREAMS_LIBRARY}
-  ${Boost_SYSTEM_LIBRARY}
   ${Tbb_TBB_LIBRARY}
   ${Ilmbase_HALF_LIBRARY}
   ${ZLIB_LIBRARY}
@@ -190,11 +192,11 @@ TARGET_LINK_LIBRARIES ( openvdb_static
   )
 TARGET_LINK_LIBRARIES ( openvdb_shared
   ${Boost_IOSTREAMS_LIBRARY}
-  ${Boost_SYSTEM_LIBRARY}
   ${Tbb_TBB_LIBRARY}
   ${Ilmbase_HALF_LIBRARY}
   ${ZLIB_LIBRARY}
   ${BLOSC_blosc_LIBRARY}
+  ${EXTRA_LIBS}
   )
 
 IF (WIN32)
@@ -225,13 +227,16 @@ ADD_EXECUTABLE ( vdb_print
   ${VDB_PRINT_SOURCE_FILES}
   )
 
+if(NOT WIN32)
+	set(EXTRA_LIBS m stdc++ dl)
+endif()
+  
 TARGET_LINK_LIBRARIES ( vdb_print
   openvdb_shared
   ${CMAKE_THREAD_LIBS_INIT}
   ${BLOSC_blosc_LIBRARY}
-  m
-  stdc++
-  )
+  ${EXTRA_LIBS}
+)
 
 SET ( VDB_RENDER_SOURCE_FILES cmd/openvdb_render/main.cc )
 SET_SOURCE_FILES_PROPERTIES ( ${VDB_RENDER_SOURCE_FILES}
@@ -249,8 +254,7 @@ TARGET_LINK_LIBRARIES ( vdb_render
   ${Openexr_ILMIMF_LIBRARY}
   ${Ilmbase_ILMTHREAD_LIBRARY}
   ${Ilmbase_IEX_LIBRARY}
-  m
-  stdc++
+  ${EXTRA_LIBS}
   )
 
 SET ( VDB_VIEW_SOURCE_FILES
@@ -265,7 +269,7 @@ SET_SOURCE_FILES_PROPERTIES ( ${VDB_VIEW_SOURCE_FILES}
   PROPERTIES
   COMPILE_FLAGS "-DOPENVDB_USE_BLOSC ${OPENVDB_USE_GLFW_FLAG} -DGL_GLEXT_PROTOTYPES=1"
   )
-IF (NOT WIN32)
+IF (FALSE)
   ADD_EXECUTABLE ( vdb_view
     ${VDB_VIEW_SOURCE_FILES}
     )
@@ -283,9 +287,8 @@ IF (NOT WIN32)
     ${GLFW_LINK_LIBRARY}
     ${GLFW_DEPENDENT_LIBRARIES}
     ${GLEW_GLEW_LIBRARY}
-    m
-    stdc++
-    )
+	${EXTRA_LIBS}
+  )
 ENDIF ()
 
 SET ( UNITTEST_SOURCE_FILES
@@ -392,8 +395,7 @@ IF ( OPENVDB_BUILD_UNITTESTS )
   TARGET_LINK_LIBRARIES ( vdb_test
     ${CPPUnit_cppunit_LIBRARY}
     openvdb_shared
-    m
-    stdc++
+    ${EXTRA_LIBS}
     )
 
   ADD_TEST ( vdb_unit_test vdb_test )
@@ -419,7 +421,7 @@ IF (OPENVDB_BUILD_DOCS)
 ENDIF ()
 
 # Installation
-IF ( NOT WIN32 )
+IF ( FALSE )
   INSTALL ( TARGETS
     vdb_view
     DESTINATION
diff --git a/openvdb/math/Coord.h b/openvdb/math/Coord.h
index f2d288ea..4eb6cb05 100644
--- a/openvdb/math/Coord.h
+++ b/openvdb/math/Coord.h
@@ -35,6 +35,7 @@
 #include <array> // for std::array
 #include <iostream>
 #include <limits>
+#define NOMINMAX
 #include <openvdb/Platform.h>
 #include "Math.h"
 #include "Vec3.h"
